### PR TITLE
Azure: Add global Storage Accounts detectors

### LINKED ISSUE #206: Cloud Azure - Storage detectors
Rework Azure Storage Account detectors to replace those in the stable branch.

Proposal: 6 new modules
* general storage account with Azure limits
* blob storage
* queue storage
* table storage
* file standard
* file premium (due to different behavior)

Claranet reference: AZ-394

### PR DESCRIPTION
**Pull request by BzSpi**
Adding global storage account detector. 
Checks:
* overall capacity limit
* resources count limit
* ingress limit
* egress limit
* transactions rate limit

fixes #206

### CODE REVIEW

**Review by Shr3ps — APPROVED** [review]

File: modules/integration_azure-storage-account/conf/01-count.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module: "Azure Storage Account"
+name: "count"
+filtering: "filter('resource_type', 'Microsoft.Storage/storageAccounts') and filter('primary_aggregation_type', 'true')"
+aggregation: ".fill(None, duration='1d').count(by=['subscription_id'])"
```
Comment (xp-1000) [inline modules/integration_azure-storage-account/conf/01-count.yaml:4]: why not simply `fill()` ?

File: modules/integration_azure-storage-account/variables-gen.tf [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+variable "capacity_notifications" {
+  description = "Notification recipients list per severity overridden for capacity detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "capacity_aggregation_function" {
+  description = "Aggregation function and group by for capacity detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".fill(None, duration='1d').sum(by=['azure_resource_name', 'azure_resource_group_name', 'azure_region'])"
```
Comment (xp-1000) [inline modules/integration_azure-storage-account/variables-gen.tf:74]: I understand why you do this. while it is not possible to use fill in current generator template you can trick it here.

I think a better approach would be to use a formula base on this simple metric signal and use fill inside this formula (and not the underlying metric signal).

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/integration_azure-storage-account/conf/01-count.yaml (hunk 1 above)
Comment (BzSpi) [reply modules/integration_azure-storage-account/conf/01-count.yaml:4]: Because I want it to stop after one day :)

Comment (xp-1000) [reply modules/integration_azure-storage-account/conf/01-count.yaml]: ok, it is not common so I would to be sure :p

**Review by xp-1000 — APPROVED** [review]

**Pull request merged**